apis/v1beta1/types.go

```diff
@@ -165,11 +165,16 @@ type VirtualMachineCloneSpec struct {
 	// PciDevices is the list of pci devices used by the virtual machine.
 	// +optional
 	PciDevices []PCIDeviceSpec `json:"pciDevices,omitempty"`
-
 	// OS is the Operating System of the virtual machine
 	// Defaults to Linux
 	// +optional
 	OS OS `json:"os,omitempty"`
+	// VMXVersion is the hardware version of the virtual machine.
+	// Defaults to the eponymous property value in the template from which the
+	// virtual machine is cloned.
+	// Check the compatibility with the ESXi version before setting the value.
+	// +optional
+	VMXVersion string `json:"vmxVersion,omitempty"`
```

Comment:
we should probably rename to `hardwareVersion`

Reply:
makes sense, done.